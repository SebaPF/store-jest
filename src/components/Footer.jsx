import React from 'react';
import '../styles/components/Footer.styl';

const Footer = () => (
  <div className="Footer">
    <p className="Footer-title">Platzi Store</p>
    <p className="Footer-copy">Hago un Cambio y Actualizo mi SNAPSHOT...</p>
  </div>
);

export default Footer;
