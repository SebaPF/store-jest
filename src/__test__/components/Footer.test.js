import React from 'react';
import { mount } from 'enzyme';
import { create } from 'react-test-renderer';
import Footer from '../../components/Footer';

// Si no existe el 'Snapshot', lo crea y luego lo comprueba y, cuando vuelvo a correr mis pruebas y ya existe, los Compara ... //
// jest --updateSnapshot // CUANDO QUIERA ACTUALIZAR MI SNAPSHOT CON NUEVOS CAMBIOS

const footer = mount(<Footer />);

describe('<Footer />', () => {
  test('Render del componente Footer', () => {
    expect(footer.length).toEqual(1);
  });

  test('Render del titulo', () => {
    expect(footer.find('.Footer-title').text()).toEqual('Platzi Store');
  });

  test('Footer Snapshot', () => {
    const foot = create(<Footer />);
    expect(foot.toJSON()).toMatchSnapshot();
  });

});
