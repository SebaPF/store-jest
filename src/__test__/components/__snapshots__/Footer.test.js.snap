// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> Footer Snapshot 1`] = `
<div
  className="Footer"
>
  <p
    className="Footer-title"
  >
    Platzi Store
  </p>
  <p
    className="Footer-copy"
  >
    Hago un Cambio y Actualizo mi SNAPSHOT...
  </p>
</div>
`;
